### ISSUE TITLE
Tunnelblick sometimes crashes when .ovpn file contains non-ASCII characters

### ISSUE THREAD

**Issue by furiousbean** [Tunnelblick/Tunnelblick#333]
I have trouble using .ovpn configurations with russian comments in UTF-8 codepage. Import of such configurations works well, but when I try to start connection/change details, Tunnelblick crashes with errors like this:
```
2016-09-22 15:21:02.969 Tunnelblick[25223:1247109] An uncaught exception was raised
2016-09-22 15:21:02.969 Tunnelblick[25223:1247109] -[__NSCFString appendString:]: nil argument
2016-09-22 15:21:02.975 Tunnelblick[25223:1247109] (
	0   CoreFoundation                      0x00007fff8ca134f2 __exceptionPreprocess + 178
	1   libobjc.A.dylib                     0x00007fff8ea97f7e objc_exception_throw + 48
	2   CoreFoundation                      0x00007fff8ca7a4bd +[NSException raise:format:] + 205
	3   CoreFoundation                      0x00007fff8ca4885f mutateError + 159
	4   Tunnelblick                         0x000000010f6b44bb runTunnelblickd + 956
	5   Tunnelblick                         0x000000010f66dea6 runOpenvpnstart + 237
	6   Tunnelblick                         0x000000010f67b63f +[ConfigurationManager parseConfigurationPath:forConnection:hasAuthUserPass:] + 199
	7   Tunnelblick                         0x000000010f6415ec -[VPNConnection setTunOrTapAndHasAuthUserPass] + 78
	8   Tunnelblick                         0x000000010f641b3a -[VPNConnection argumentsForOpenvpnstartForNow:userKnows:] + 387
	9   Tunnelblick                         0x000000010f640a2a -[VPNConnection connect:userKnows:] + 1308
	10  libsystem_trace.dylib               0x00007fff8cdfd07a _os_activity_initiate + 75
	11  AppKit                              0x00007fff8d857dbd -[NSApplication sendAction:to:from:] + 460
	12  AppKit                              0x00007fff8d857b57 -[NSMenuItem _corePerformAction] + 336
	13  AppKit                              0x00007fff8d8578b7 -[NSCarbonMenuImpl performActionWithHighlightingForItemAtIndex:] + 114
	14  libsystem_trace.dylib               0x00007fff8cdfd07a _os_activity_initiate + 75
	15  AppKit                              0x00007fff8d8e9e54 -[NSMenu performActionForItemAtIndex:] + 131
	16  AppKit                              0x00007fff8d8e9dc7 -[NSMenu _internalPerformActionForItemAtIndex:] + 35
	17  AppKit                              0x00007fff8d8e9c1f -[NSCarbonMenuImpl _carbonCommandProcessEvent:handlerCallRef:] + 107
	18  AppKit                              0x00007fff8d78efd9 NSSLMMenuEventHandler + 708
	19  HIToolbox                           0x00007fff8440a7be _ZL23DispatchEventToHandlersP14EventTargetRecP14OpaqueEventRefP14HandlerCallRec + 1231
	20  HIToolbox                           0x00007fff84409c48 _ZL30SendEventToEventTargetInternalP14OpaqueEventRefP20OpaqueEventTargetRefP14HandlerCallRec + 404
	21  HIToolbox                           0x00007fff8441f9e6 SendEventToEventTarget + 40
	22  HIToolbox                           0x00007fff8446999a _ZL18SendHICommandEventjPK9HICommandjjhPKvP20OpaqueEventTargetRefS5_PP14OpaqueEventRef + 411
	23  HIToolbox                           0x00007fff84494d5b SendMenuCommandWithContextAndModifiers + 59
	24  HIToolbox                           0x00007fff84494d0c SendMenuItemSelectedEvent + 188
	25  HIToolbox                           0x00007fff84494be8 _ZL19FinishMenuSelectionP13SelectionDataP10MenuResultS2_ + 96
	26  HIToolbox                           0x00007fff84473b81 _ZL19PopUpMenuSelectCoreP8MenuData5PointdS1_tjPK4RecttjS4_S4_PK14__CFDictionaryPK10__CFStringPP13OpaqueMenuRefPt + 1794
	27  HIToolbox                           0x00007fff84472c5c _ZL26_HandlePopUpMenuSelection8P13OpaqueMenuRefP14OpaqueEventRefj5PointtjPK4RecttS6_S6_PK14__CFDictionaryPK10__CFStringPS0_Pt + 610
	28  HIToolbox                           0x00007fff84472853 _HandlePopUpMenuSelectionWithDictionary + 287
	29  AppKit                              0x00007fff8d8e0c6c _NSSLMPopUpCarbonMenu3 + 6168
	30  AppKit                              0x00007fff8d8df442 _NSPopUpCarbonMenu3 + 153
	31  AppKit                              0x00007fff8d8df39a -[NSCarbonMenuImpl popUpMenu:atLocation:width:forView:withSelectedItem:withFont:withFlags:withOptions:] + 356
	32  AppKit                              0x00007fff8dcc6ef9 +[NSStatusBarButtonCell popupStatusBarMenu:inRect:ofView:withEvent:] + 705
	33  AppKit                              0x00007fff8dccae7d -[NSStatusItem popUpStatusItemMenu:] + 173
	34  Tunnelblick                         0x000000010f6a97de -[MainIconView mouseDownMainThread:] + 206
	35  Foundation                          0x00007fff848d2fde __NSThreadPerformPerform + 279
	36  CoreFoundation                      0x00007fff8c9a8881 __CFRUNLOOP_IS_CALLING_OUT_TO_A_SOURCE0_PERFORM_FUNCTION__ + 17
	37  CoreFoundation                      0x00007fff8c987fbc __CFRunLoopDoSources0 + 556
	38  CoreFoundation                      0x00007fff8c9874df __CFRunLoopRun + 927
	39  CoreFoundation                      0x00007fff8c986ed8 CFRunLoopRunSpecific + 296
	40  HIToolbox                           0x00007fff84432935 RunCurrentEventLoopInMode + 235
	41  HIToolbox                           0x00007fff84432677 ReceiveNextEventCommon + 184
	42  HIToolbox                           0x00007fff844325af _BlockUntilNextEventMatchingListInModeWithFilter + 71
	43  AppKit                              0x00007fff8d5eedf6 _DPSNextEvent + 1067
	44  AppKit                              0x00007fff8d5ee226 -[NSApplication _nextEventMatchingEventMask:untilDate:inMode:dequeue:] + 454
	45  AppKit                              0x00007fff8d5e2d80 -[NSApplication run] + 682
	46  AppKit                              0x00007fff8d5ac368 NSApplicationMain + 1176
	47  Tunnelblick                         0x000000010f63c558 start + 52
)
```

"Sometimes" means that in some cases Tunnelblick works OK with UTF-8 .ovpn configurations, but even change of domain name (which contain _only_ latin symbols) can lead to crash.

**Workaround:** remove all non-ASCII symbols from .ovpn configuration file.

**Comment by jkbullard** [comment]
@furiousbean - 

Thank you for reporting this.

 1. What version of Tunnelblick are you using?

 2. Could you post an example of a configuration file that fails this way so it can be tested? (The configuration does not need to actually work to create a VPN because this error happens before starting OpenVPN, when Tunnelblick is trying to parse the configuration file.)

**Comment by furiousbean** [comment]
@jkbullard -

1. Both 3.6.5 and latest stable 3.6.7_build_4602 versions have this issue.
2. Yep, here is two configurations, Tunnelblick crashes on ["crashes.ovpn"](http://pastebin.com/3M7VdN7a) and does not on ["doesnotcrash.ovpn"](http://pastebin.com/nBhJRLvH).

**Comment by jkbullard** [comment]
@furiousbean -

Thank you. I will look into this.

**Comment by jkbullard** [comment]
@furiousbean -

I can't reproduce this error, but examining the code I can see what I think the problem is and will fix it as soon as I can, hopefully within the next few days.

**Comment by furiousbean** [comment]
OK, thank you. I'll check the fixed version and reply here if issue will still present.

**Comment by jkbullard** [comment]
@furiousbean - Thanks again for reporting this and helping diagnose the problem.

A fix for this has been committed (b85cb51) and will be included in the next beta release.

Note: In addition to causing problems when connecting, this bug could cause problems when a configuration is selected in the left side of the Configurations tab of the VPN Details window.
